http-client-core/src/main/java/io/micronaut/http/client/interceptor/HttpClientIntroductionAdvice.java

```diff
@@ -546,6 +469,33 @@ protected void doOnComplete() {
         return context.proceed();
     }
 
+    private Object getValue(Argument argument, MethodInvocationContext<?, ?> context) {
+        String argumentName = argument.getName();
+        AnnotationMetadata argumentMetadata = argument.getAnnotationMetadata();
+        MutableArgumentValue<?> value = context.getParameters().get(argumentName);
+
+        Object definedValue = value.getValue();
+        Map<String, Object> paramMap = context.getParameterValueMap();
```

Comment:
`context.getParameterValueMap()` should not be called repeatedly. It is an expensive operation from a performance perspective